x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,146 @@
+		destinationHoldingAddress = hostZone.CommunityPoolStakeHoldingAddress
+	case stIbcDenom:
+		destinationHoldingAddress = hostZone.CommunityPoolRedeemHoldingAddress
+	default:
+		return fmt.Errorf("Invalid community pool transfer denom: %s", token.Denom)
+	}
+
+	memo := ""
+	var msgs []proto.Message
+	msgs = append(msgs, transfertypes.NewMsgTransfer(
+		transfertypes.PortID,
+		counterpartyChannelId, // for transfers of communityPoolHostZone -> Stride
+		token,
+		hostZone.CommunityPoolDepositIcaAddress, // ICA controlled address on community pool zone
+		destinationHoldingAddress,               // Stride address, unique to each community pool / hostzone
+		clienttypes.Height{},
+		endEpochTimestamp,
+		memo,
+	))
+
+	// No need to build ICA callback data or input an ICA callback method since the callback Stride can see is only
+	//  the ICA callback, not the actual transfer callback. The transfer ack returns to the hostZone chain not Stride
+	icaCallbackId := ""
+	var icaCallbackData []byte
+
+	// Send the transaction through SubmitTx to kick off ICA commands -- no ICA callback method name, or callback args needed
+	_, err = k.SubmitTxs(ctx,
+		hostZone.ConnectionId,
+		msgs,
+		types.ICAAccountType_COMMUNITY_POOL_DEPOSIT,
+		endEpochTimestamp,
+		icaCallbackId,
+		icaCallbackData)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Failed to SubmitTxs, Messages: %v, err: %s", msgs, err.Error())
+	}
+	k.Logger(ctx).Info("Successfully sent ICA command to kick off ibc transfer from deposit ICA to stake holding address")
+
+	return nil
+}
+
+// Transfers a recently minted stToken from the stride-side stake holding address to the return ICA address on the host zone
+func (k Keeper) TransferHoldingToCommunityPoolReturn(ctx sdk.Context, hostZone types.HostZone, coin sdk.Coin) error {
+	memo := ""
+	strideEpochTracker, found := k.GetEpochTracker(ctx, epochstypes.STRIDE_EPOCH)
+	if !found {
+		return errorsmod.Wrapf(types.ErrEpochNotFound, epochstypes.STRIDE_EPOCH)
+	}
+	endEpochTimestamp := uint64(strideEpochTracker.NextEpochStartTime)
+
+	// build and send an IBC message for each coin to transfer all back to the hostZone
+	msg := transfertypes.NewMsgTransfer(
+		transfertypes.PortID,
+		hostZone.TransferChannelId,
+		coin,
+		hostZone.CommunityPoolStakeHoldingAddress, // from Stride address, unique to each community pool / hostzone
+		hostZone.CommunityPoolReturnIcaAddress,    // to ICA controlled address on foreign hub
+		clienttypes.Height{},
+		endEpochTimestamp,
+		memo,
+	)
+
+	msgTransferResponse, err := k.RecordsKeeper.TransferKeeper.Transfer(sdk.WrapSDKContext(ctx), msg)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Error submitting ibc transfer for %+v", coin)
+	}
+
+	result := fmt.Sprintf("Successfully submitted ibctransfer for %+v with response %+v",
+		coin, msgTransferResponse)
+	k.Logger(ctx).Info(result)
+
+	return nil
+}
+
+// given a hostZone with native denom, returns the ibc denom on the zone for the staked stDenom
```

Comment:
Please add unit tests for this, especially the counterparty channel logic

Reply:
There’s a test [here](https://github.com/Stride-Labs/stride/blob/d0e01c74ab7b7a375bfa45bebd764197924a720b/x/stakeibc/keeper/transfer_test.go#L175) already - lmk if you were picturing something else!

Reply:
Missed that one, nice! Resolving